src/api/CoinBase.ts

```diff
@@ -0,0 +1,63 @@
+import { ExchangeApi } from '../types/ExchangeApi';
+import fetch, { Response } from 'node-fetch';
+import BigNumber from 'bignumber.js';
+
+const URL = 'https://api.coinbase.com/v2/exchange-rates';
+
+export type CoinBaseResponse = {
+    data: {
+        currency: string;
+        rates: Record<string, string>;
+    };
+};
+
+const CURRENCY_MAPPING: Record<string, string> = {
+    RIF: 'RIF',
+    RBTC: 'RBTC',
+    TKN: 'RIF'
+};
+
+export class CoinBase implements ExchangeApi {
+    getApiTokenName(tokenSymbol: string): string {
+        if (!tokenSymbol) {
+            throw Error('ExchangeApi cannot map a token with a null value');
+        }
+        const resultMapping = CURRENCY_MAPPING[tokenSymbol];
```

Comment:
Have you tested this with lower/upper case tokenSymbol?

Reply:
updated